optuna/notebooks/optuna_rapids.ipynb

```diff
@@ -0,0 +1,687 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Score with default parameters :  0.7608551383018494\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Score with default parameters : \",train_and_eval(X, y))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Objective Function\n",
+    " \n",
+    "The objective function will be the one we optimize in [Optuna Study](https://optuna.readthedocs.io/en/stable/reference/study.html). Objective function tries out different values for the parameters that we are tuning and saving the results in `study.trials_dataframes()`. \n",
+    " \n",
+    "Let's define the objective function for this HPO task by making use of the `train_and_eval()`. You can see that we simply choose a value for the parameters and call the `train_and_eval` method, making Optuna very easy to use in an existing workflow.\n",
+    " \n",
+    "The objective remains constant over different [samplers](https://optuna.readthedocs.io/en/stable/reference/samplers.html), which are built-in options in Optuna to enable the selection of different sampling algorithms that optuna provides. Some of the available ones include - GridSampler, RandomSampler, TPESampler, etc. We'll use TPESampler for this demo, but feel free to try different samplers to notice the changes in performance. \n",
+    " \n",
+    "\n",
+    "[Tree-Structured Parzen Estimators](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.TPESampler.html#optuna.samplers.TPESampler) or TPE works by fitting two  Gaussian Mixture Model during each trial - one to the set of parameter values associated with the best objective values,\n",
+    "and another to the remaining parameter values. It chooses the parameter value that maximizes the ratio between the two GMMs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def objective(trial, X_param, y_param):\n",
+    "    max_depth = trial.suggest_int(\"max_depth\", 7, 15)\n",
+    "    n_estimators = trial.suggest_int(\"n_estimators\", 100, 1000)\n",
+    "    score = train_and_eval(X_param, y_param, max_depth=max_depth,\n",
+    "                           n_estimators=n_estimators)\n",
+    "    return score"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## HPO Trials and Study\n",
+    " \n",
+    "Optuna uses [study](https://optuna.readthedocs.io/en/stable/reference/study.html) and [trials](https://optuna.readthedocs.io/en/stable/reference/trial.html) to keep track of the HPO experiments. Put simply, a trial is a single call of the objective function while a set of trials make up a study. We will pick the optimal performing trial from a study to get the best parameters that were used in that run.\n",
```

Comment:
I'm always hesitant to use the word "optimal" since it's not provably optimal... maybe "best observed trial"